src/store/products/sagas.js

```diff
@@ -0,0 +1,44 @@
+import { takeEvery, put, all } from 'redux-saga/effects';
+import { setProducts, setOneProduct, GET_PRODUCTS_ASYNC, GET_ONE_PRODUCT_ASYNC, UPDATE_ONE_PRODUCT_ASYNC, DELETE_ONE_PRODUCT_ASYNC, SEND_ONE_PRODUCT_ASYNC, getProductsAsync, SEND_IMAGE_ASYNC } from './actions';
+import { getProducts, getProduct, requestUpdateProduct, requestDeleteProduct, sendProduct, sendImage } from '../../services';
+
+function* getProductsHandler() {
+  const products = yield getProducts();
+  yield put(setProducts(products));
+}
+
+function* getProductHandler(action) {
+  const product = yield getProduct(action.data);
+  yield put(setOneProduct(product));
+}
+
+function* updateProductHandler(action) {
+  yield requestUpdateProduct(action.data);
+  yield put(getProductsAsync());
```

Comment:
when you are on the product page and update current product, why need to get all products?